### PR TITLE
API metrics

### PR DESCRIPTION
**Pull request by taminomara**
Re-implement API metrics using the new metric storage

### Test Plan
CI and manual testing

### CODE REVIEW

File: orderbook/src/api.rs [hunk 1]
```diff
@@ -41,66 +36,62 @@ pub fn handle_all_routes(
...
-                .map(|reply| LabelledReply::new(reply, "get_fee_and_quote_sell")))
-            .unify()
-            .or(get_fee_and_quote_buy
-                .map(|reply| LabelledReply::new(reply, "get_fee_and_quote_buy")))
-            .unify()
-            .or(get_user_orders.map(|reply| LabelledReply::new(reply, "get_user_orders")))
-            .unify(),
+        (create_order.with(handle_metrics("create_order")))
+            .or(get_orders.with(handle_metrics("get_orders")))
+            .or(fee_info.with(handle_metrics("fee_info")))
```
Comment (bh2smith) [inline orderbook/src/api.rs:52]: This looks much nicer, could you elaborate on how the api metrics change is related to this. Or even why we no longer need "unify"?

File: orderbook/src/api.rs [hunk 2]
```diff
@@ -41,66 +36,62 @@ pub fn handle_all_routes(
...
+#[metric(subsystem = "api", labels("endpoint"))]
+struct ApiMetrics {
+    /// Number of completed API requests.
+    #[metric(labels("status_code"))]
+    requests_complete: prometheus::CounterVec,
+
+    /// Execution time for each API request.
+    #[metric(labels())]
+    requests_duration_seconds: prometheus::Histogram,
 }
```
Comment (bh2smith) [inline orderbook/src/api.rs:95]: I see that this parameter `metrics: Arc<Metrics>,` was removed from several places, but don't see anything put back to replace it. So at this point I can only assume that this large green block is compensating for all of that. Could you please explain how the code here is instantiated in places where metrics was removed?

**Review by bh2smith — APPROVED** [review]

File: orderbook/src/api.rs (hunk 2 above)
Comment (taminomara) [reply orderbook/src/api.rs:95]: So we've switched to the global registry that's available through the `get_metric_storage_registry` function. So now we don't have to explicitly pass registry around.

**Review by nlordell — APPROVED** [review]

**Comment by nlordell** [comment]
The proc macro looks very nice to use.

**Review by vkgnosis — APPROVED** [review]

**Pull request merged**